packages/worker/src/api/controllers/global/users.ts

```diff
@@ -185,9 +186,23 @@ export const getAppUsers = async (ctx: Ctx<SearchUsersRequest>) => {
 export const search = async (ctx: Ctx<SearchUsersRequest>) => {
   const body = ctx.request.body
 
-  // TODO: for now only one supported search key, string.email
-  if (body?.query && !userSdk.core.isSupportedUserSearch(body.query)) {
-    ctx.throw(501, "Can only search by string.email or equal._id")
+  // TODO: for now only two supported search keys; string.email and equal._id
+  if (body?.query) {
+    // Clean numeric prefixing. This will overwrite duplicate search fields,
+    // but this is fine because we only support a single custom search on
+    // email and id
+    for (let filters of Object.values(body.query)) {
+      if (filters && typeof filters === "object") {
+        for (let [field, value] of Object.entries(filters)) {
+          delete filters[field]
+          filters[removeKeyNumbering(field)] = value
+        }
+      }
+    }
+    // Validate we aren't trying to search on any illegal fields
+    if (!userSdk.core.isSupportedUserSearch(body.query)) {
+      ctx.throw(501, "Can only search by string.email or equal._id")
```

Comment:
501 isn't the correct status code here. It's for when a specific HTTP verb hasn't been implemented by a route (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501). I think 400 is more appropriate here.

Reply:
Also would be great to have a test that hits this condition.

Reply:
I didn't write this original feature (I'm just adjusting it to account for the numeric prefixing) but I'm happy to change that status code over.

There already is a test that hits the error case from the original implementation. The 2 tests I've added are just to specifically test the numeric prefixing removal but there already are a few existing tests, including one that hits the 501 path.

Reply:
Changed to 400 👍